File: app/src/androidTest/java/de/westnordost/streetcomplete/quests/opening_hours/WeekdaysTest.java
package de.westnordost.streetcomplete.quests.opening_hours;

import junit.framework.TestCase;

public class WeekdaysTest extends TestCase
{
	// hihi
	private static final boolean l = true;
	private static final boolean o = false;

	public void testToString() {
		assertEquals("Mo,Tu", new Weekdays(new boolean[]{l,l,o,o,o,o,o,o}).toString());
		assertEquals("Tu-Th", new Weekdays(new boolean[]{o,l,l,l,o,o,o,o}).toString());
		assertEquals("Tu-Th,Sa,Su,PH", new Weekdays(new boolean[]{o,l,l,l,o,l,l,l}).toString());
		assertEquals("Su,Mo", new Weekdays(new boolean[]{l,o,o,o,o,o,l,o}).toString());
		assertEquals("Sa-Th,PH", new Weekdays(new boolean[]{l,l,l,l,o,l,l,l}).toString());
	}
}

File: app/src/test/java/de/westnordost/streetcomplete/osm/opening_hours/model/WeekdaysTest.kt
package de.westnordost.streetcomplete.osm.opening_hours.model

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class WeekdaysTest {

    @Test fun isSelectionEmpty() {
        assertTrue(weekdays(0b00000000).isSelectionEmpty())
        assertFalse(weekdays(0b00010000).isSelectionEmpty())
    }

    @Test fun `toString works`() {
        assertEquals("Mo,Tu", weekdays(0b11000000).toString())
        assertEquals("Tu-Th", weekdays(0b01110000).toString())
        assertEquals("Tu-Th,Sa,Su,PH", weekdays(0b01110111).toString())
        assertEquals("Su,Mo", weekdays(0b10000010).toString())
        assertEquals("Sa-Th,PH", weekdays(0b11110111).toString())
    }
}
